util/src/length_prefix_encoding/encoder.rs

```diff
@@ -92,6 +92,90 @@ pub struct WriteToIoReturn {
 }
 
 #[derive(Clone, Copy, Debug)]
+/// Module description goes here (skipped since it already exists in #479)
+///
+/// # Examples
+///
+///  Unfinished/proof-of-concept versions below - don't try this at home
+///
+/// **TODOs**:
+///
+/// * Actually write to a sink, if that's desirable/integration tests don't suffice
+/// * Consider splitting into multiple examples in that case
+/// * Resolve all TBD notes, fixup/polish, add references/links, etc.
+/// * rustfmt ignores docstrings, check [unstable option](https://github.com/rust-lang/rustfmt/issues/3348) or fix manually?
+/// * Decide on the most important use cases vs. per-function examples (intended API usage?)
```

Comment:
Lets settle on one use case, the integration test writing to a memory buffer above.